.github/workflows/scala.yml

```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java: [11, 21]
+        java: [17, 21]
```

Comment:
As a follow-up to my remark in https://github.com/scalatra/scalatra/pull/1610/files#r1438577326, would you know a way to to run `sbt test` for a specific (i.e. `javax` or `jakarta`) build only?
So that the `javax` build would tested against Java 8, 11, 17 and 21, while the `jakarta` build would be tested against Java 17 and 21 only.

Reply:
I had an idea: what about enabling the `jakarta` axis in `build.sbt` only in case SBT is running in a JDK 17+ environment?

Reply:
As I commented above, I think we can target Java 17+ in Scalatra 3.1.x if we would upgrade Jetty to 12, and keeping 3.0.x maintained for users who still need Java 11.

Reply:
Oh, I thought you didn't want to maintain the 3.0.x branch.
But I think I have a solution to my original question:
> So that the `javax` build would tested against Java 11, 17 and 21, while the `jakarta` build would be tested against Java 17 and 21 only.

I have implemented a "trick" in [Run tests for "javax" build on JDK 11 (only)](https://github.com/scalatra/scalatra/pull/1610/commits/cb9d2adc0184d7fffc0a771e24650d729131cc77) in order to implement that.
What do you think?

Reply:
> Oh, I thought you didn't want to maintain the 3.0.x branch.

Haha. Yeah, I don't want to maintain multiple branches if possible, but we always had to do it because when we upgrade Scalatra version these days, there were always incompatible changes like dropping Scala version or Java version. We needed to maintain old version for users who need it.

This time, I only think that dropping `javax` is too early.

Reply:
Up to you to decide 😄 
I shall revert/leave the code as you wish.

Reply:
For the record, I've restricted the use of the `runningOnJvm17AndAbove()` check to the `jetty` module when used in the `jakarta` build.

Reply:
I stand correct on my last comment. It's also the `test` module that has to be excluded when running on Java 11, because this module internally relies on Jetty via `[Embedded]JettContainer`.

I suppose we could introduce a `[Embedded]TomcatContainer` and use it when the test suite is run on Java 11?
But I suppose that would require some "magic" (macro?) to dynamically choose between Jetty and Tomcat.
Is that an option that is worth exploring?

Reply:
Hmm, right. Or simply using `EmbeddedTomcatContainer` in tests could be an option. But again, I don't mind about dropping Java 11 support in Scalatra 3.1.